keras/preprocessing/sequence.py

```diff
@@ -35,7 +35,14 @@ def pad_sequences(sequences, maxlen=None, dtype='int32', padding='pre', truncati
     if maxlen is None:
         maxlen = np.max(lengths)
 
-    x = (np.ones((nb_samples, maxlen)) * value).astype(dtype)
+    # assuming sample shapes are consistent in `sequences` - take the first one that is not empty
```

Comment:
This assumption should be checked. Users will throw all kinds of malformed data at your code, and you need to be able to raise helpful error messages in these cases.